<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="SophieFournier" />

<meta name="date" content="2020-12-07" />

<title>first-analysis</title>

<script src="site_libs/header-attrs-2.5/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/SophieFournier/myproject">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">first-analysis</h1>
<h4 class="author">SophieFournier</h4>
<h4 class="date">2020-12-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>myproject/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201207code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201207)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201207code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201207)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomSophieFourniermyprojecttree3ee55ef4a870c5f9061f0ae7babb0b9a75753521targetblank3ee55efa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/SophieFournier/myproject/tree/3ee55ef4a870c5f9061f0ae7babb0b9a75753521" target="_blank">3ee55ef</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomSophieFourniermyprojecttree3ee55ef4a870c5f9061f0ae7babb0b9a75753521targetblank3ee55efa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/SophieFournier/myproject/tree/3ee55ef4a870c5f9061f0ae7babb0b9a75753521" target="_blank">3ee55ef</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/Rare-Variant Association Methods.Rmd
    Untracked:  site_libs/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/first-analysis.Rmd</code>) and HTML (<code>docs/first-analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SophieFournier/myproject/blob/3ee55ef4a870c5f9061f0ae7babb0b9a75753521/analysis/first-analysis.Rmd" target="_blank">3ee55ef</a>
</td>
<td>
SophieFournier
</td>
<td>
2021-01-28
</td>
<td>
Add my first analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SophieFournier/myproject/d5040daeb057dfa0fd94ebf6da21f76b987196e0/docs/first-analysis.html" target="_blank">d5040da</a>
</td>
<td>
SophieFournier
</td>
<td>
2021-01-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SophieFournier/myproject/blob/a2b4430dbf2b1a8691083099f4033dcf4b13d5bc/analysis/first-analysis.Rmd" target="_blank">a2b4430</a>
</td>
<td>
SophieFournier
</td>
<td>
2021-01-26
</td>
<td>
Add my first analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SophieFournier/myproject/b10461814799166ad768126494ddb0c0c6248a12/docs/first-analysis.html" target="_blank">b104618</a>
</td>
<td>
SophieFournier
</td>
<td>
2021-01-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SophieFournier/myproject/blob/d656ac1691be89da715b90356d2ffec812cd1832/analysis/first-analysis.Rmd" target="_blank">d656ac1</a>
</td>
<td>
SophieFournier
</td>
<td>
2021-01-26
</td>
<td>
Add my first analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SophieFournier/myproject/8e90e218c563cb5386c8d88fa9bc2915bcd451ba/docs/first-analysis.html" target="_blank">8e90e21</a>
</td>
<td>
SophieFournier
</td>
<td>
2021-01-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SophieFournier/myproject/blob/aaea4a25c15dfc5e2cbbc8868348fefb933401fc/analysis/first-analysis.Rmd" target="_blank">aaea4a2</a>
</td>
<td>
SophieFournier
</td>
<td>
2021-01-05
</td>
<td>
Add my first analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SophieFournier/myproject/b18ad430facf198aa7f4296e29520bff715b2b28/docs/first-analysis.html" target="_blank">b18ad43</a>
</td>
<td>
SophieFournier
</td>
<td>
2020-12-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SophieFournier/myproject/blob/e36bfca26494166ff11acba6d477d947f2c2e2f8/analysis/first-analysis.Rmd" target="_blank">e36bfca</a>
</td>
<td>
SophieFournier
</td>
<td>
2020-12-23
</td>
<td>
Add my first analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SophieFournier/myproject/84489affad7519f6d6ae9d0a27df94c5c5a8a1ab/docs/first-analysis.html" target="_blank">84489af</a>
</td>
<td>
SophieFournier
</td>
<td>
2020-12-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SophieFournier/myproject/a2151d6b2a091d211a8843cd59f646b91a6698d3/docs/first-analysis.html" target="_blank">a2151d6</a>
</td>
<td>
SophieFournier
</td>
<td>
2020-12-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SophieFournier/myproject/blob/422789e3bb75c3134ca05640424f9cde905a886a/analysis/first-analysis.Rmd" target="_blank">422789e</a>
</td>
<td>
SophieFournier
</td>
<td>
2020-12-23
</td>
<td>
Add my first analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SophieFournier/myproject/6258b205b335857e38727f4561fa67888ef15dd4/docs/first-analysis.html" target="_blank">6258b20</a>
</td>
<td>
SophieFournier
</td>
<td>
2020-12-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SophieFournier/myproject/blob/0f3609f89ee0fc219f940168e78728e292e87146/analysis/first-analysis.Rmd" target="_blank">0f3609f</a>
</td>
<td>
SophieFournier
</td>
<td>
2020-12-07
</td>
<td>
Add my first analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="rare-variant-association-methods" class="section level2">
<h2>Rare-Variant Association Methods</h2>
<p>Burden tests: <strong>bold</strong> <strong>### Header 3</strong></p>
<p>Strengths – are more powerful when there is a large proportion of variants are casual and effects are in the same direction</p>
<p>Limitations – less powerful in the presence of both trait increasing and decreasing variants or a small proportion of causal variants are often</p>
<p>CMC <strong>block quote</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Idea: that uses allele frequencies to determine the divisions of the variants into groups. After the rare are selected, they are collapsed into an indicator variable and then a multivariate test is applied to the collection formed by the common variants and the collapsed variant.</p></li>
<li><p>Model: For the CMC method, markers are divided into subgroups. (e.g. allele frequencies), and within each group, marker data are collapsed. A multivariate test (e.g. Hotelling’s T2 test) is then applied for analysis of the groups of marker data.</p></li>
<li><p>R function</p></li>
</ol>
<p>Simple marker test equation:</p>
<p>Multiple-marker test equation:</p>
<p>Hotelling’s T2 test:</p>
<p>CMC(y, X, maf = 0.05, perm = 100)</p>
<p>Y: Numeric vector with phenotype status: 0=controls, 1=cases. No missing data allowed.</p>
<p>X: numeric matrix or data frame with genotype data coded as 0, 1, 2. Missing data is allowed.</p>
<p>Adaptive Burden tests: <strong>bold</strong> <strong>### Header 3</strong></p>
<p>Strengths – are stronger than burden tests using fixed weights or thresholds. Some tests can improve result interpretation</p>
<p>Limitations – are usually computationally intensive. VT requires the same assumptions as burden tests</p>
<p>aSum test <strong>block quote</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Idea: The aSum test is also called an Adaptive sum test and it is the approach behind the adaptive test is to use the U-statistics of the score test in order to construct a statistic with the first components of the score vector U.</p></li>
<li><p>Model: The Sum test can be used to summarize information on multiple Nucleotide Polymorphisms (SNP’s) with a DF=1. Adaptive Sum tests are low power and have limited control over error type I.</p></li>
<li><p>R function</p></li>
</ol>
<p>Equation:</p>
<p>ASUM(y, X, perm = 100)</p>
<p>Y: numeric vector with phenotype status: 0=controls, 1=cases. No missing data allowed</p>
<p>X: numeric matrix or data frame with genotype data coded as 0, 1, 2. Missing data is allowed</p>
<p>Variance-Component tests: <strong>bold</strong> <strong>### Header 3</strong></p>
<p>Strengths – are powerful in the presence of both trait increasing and decreasing variants or a small amount of causal variants</p>
<p>Limitations – are less powerful than burden tests when most variants are causal and effects are in the same direction</p>
<p>SKAT <strong>block quote</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Idea: SKAT is a Sequencing Kernal Association Test which can be used to calculate p-values by fitting the null mode containing only the covariates and so it can easily be applied to genome-wide data.</p></li>
<li><p>Model: SKAT tests have a relationship between variants and the phenotype by using classical multiple linear and logistic regression.</p></li>
<li><p>R function</p></li>
</ol>
<p>Equation:</p>
<p>SKAT(y, X, kernel = “linear”, weights = NULL, a = 1, b = 25, perm = NULL)</p>
<p>Y: numeric vector with phenotype status: 0=controls, 1=cases. No missing data allowed</p>
<p>X: numeric matrix or data frame with genotype data coded as 0, 1, 2.</p>
<p>C-alpha test <strong>block quote</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Idea: C-alpha tests allow for consistent and/or increased power and straightforward calculations when testing the variance of a set containing both protective and risk variants.</p></li>
<li><p>Model:</p></li>
<li><p>R function</p></li>
</ol>
<p>Binomial distribution equation:</p>
<p>CALPHA(y, X, perm = NULL)</p>
<p>Y: numeric vector with phenotype status: 0=controls, 1=cases. No missing data allowed</p>
<p>X: numeric matrix or data frame with genotype data coded as 0, 1, 2.</p>
<p>Combined tests: <strong>bold</strong> <strong>### Header 3</strong></p>
<p>Strengths – are stronger with respect to the percentage of causal variants and the presence of both trait increasing and decreasing variants.</p>
<p>Limitations – can be slightly less powerful than burden or variance-component tests if their assumptions are largely held; some methods are computationally intensive like the Fisher method</p>
<p>SKAT-O <strong>block quote</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Idea: SKAT-O is part of the extended family of SKAT and it is optimal unified test. This test doesn’t need to be resampled when computing asymptotic p values and it is easy to adjust to covariates.</p></li>
<li><p>Model:</p></li>
<li><p>R function</p></li>
</ol>
<p>Equation:</p>
<p>Implementation</p>
<p>Y:</p>
<p>X:</p>
<p>EC test: <strong>bold</strong> <strong>### Header 3</strong></p>
<p>Strengths – is powerful when a very small proportion of variants are causal</p>
<p>Limitations – is computationally intensive; is less powerful when a moderate or large proportion of variants are causal</p>
<p>EC test <strong>block quote</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Idea:</p></li>
<li><p>Model:</p></li>
<li><p>R function</p></li>
</ol>
<p>Equation:</p>
<p>Implementation</p>
<p>Y:</p>
<p>X:</p>
<p>Sources: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4085641/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4085641/</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2842185/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2842185/</a></p>
<p><a href="https://www.rdocumentation.org/packages/AssotesteR/versions/0.1-10/topics/CMC" class="uri">https://www.rdocumentation.org/packages/AssotesteR/versions/0.1-10/topics/CMC</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2912645/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2912645/</a></p>
<p><a href="https://rdrr.io/cran/AssotesteR/man/ASUM.html" class="uri">https://rdrr.io/cran/AssotesteR/man/ASUM.html</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3135811/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3135811/</a></p>
<p><a href="https://rdrr.io/cran/AssotesteR/man/SKAT.html" class="uri">https://rdrr.io/cran/AssotesteR/man/SKAT.html</a></p>
<p><a href="https://rdrr.io/cran/AssotesteR/man/CALPHA.html" class="uri">https://rdrr.io/cran/AssotesteR/man/CALPHA.html</a></p>
<p><a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1001322" class="uri">https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1001322</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.7 LTS

Matrix products: default
BLAS:   /usr/lib/atlas-base/atlas/libblas.so.3.0
LAPACK: /usr/lib/atlas-base/atlas/liblapack.so.3.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5      rprojroot_2.0.2 digest_0.6.27   later_1.1.0.1  
 [5] R6_2.5.0        git2r_0.27.1    magrittr_2.0.1  evaluate_0.14  
 [9] stringi_1.5.3   rlang_0.4.9     fs_1.5.0        promises_1.1.1 
[13] whisker_0.4     rmarkdown_2.5   tools_4.0.3     stringr_1.4.0  
[17] glue_1.4.2      httpuv_1.5.4    xfun_0.19       yaml_2.2.1     
[21] compiler_4.0.3  htmltools_0.5.0 knitr_1.30     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
