---
title: "first-analysis"
author: "SophieFournier"
date: "2020-12-07"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Rare-Variant Association Methods

```{r}

```
##### **Burden tests:**

Strengths – are more powerful when there is a large proportion of variants are casual and effects are in the same direction

Limitations – less powerful in the presence of both trait increasing and decreasing variants or a small proportion of causal variants are often

* **CMC**

(a) Idea: CMC method, also known as Combined Multivariate and Collapsing Method, uses allele frequencies to determine the divisions of the variants into groups. After the rare are selected, they are collapsed into an indicator variable and then a multivariate test is applied to the collection formed by the common variants and the collapsed variant.

(b) Model: For the CMC method, markers are divided into subgroups. (e.g. allele frequencies), and within each group, marker data are collapsed. A multivariate test (e.g. Hotelling’s T2 test) is then applied for analysis of the groups of marker data. 

(c) R function

Single-marker test equation:

$$X_i^2 = N\frac{[p_i^D(aa) - p_i^N(aa)]^2}{p_i^D(aa)+p_i^N(aa)} + \frac{[p_i^D(Aa)+p_i^D(AA) - p_i^N(As) - p_i^N(AA)]^2}{p_i^D(Aa) +p_i^D(AA) + p_i^N(Aa) + p_i^N(AA)}$$

Multiple-marker test equation:
  
$$S = \frac{1}{N_A + N_A - 2}{{\sum_{j = 1}^{N_A}}{(x_j - \bar{x})(x_j - \bar{x})^T} + {\sum_{j = 1}^{N_\bar{A}}{(Y_j - \bar{Y})(Y_j - \bar{Y})^T}}} $$

Hotelling's T^2 test:

$$T^2 = \frac{N_AN_\bar{A}}{N_A + N_\bar{A}}(\bar{X} - \bar{Y})^TS^{-1}(\bar{X} - \bar{Y})$$

Y: Numeric vector with phenotype status: 0=controls, 1=cases. No missing data allowed.

X: numeric matrix or data frame with genotype data coded as 0, 1, 2. Missing data is allowed.

(d) Example

 # number of cases

 cases = 500

 # number of controls

 controls = 500

 # total (cases + controls)

 total = cases + controls

 # phenotype vector

 phenotype = c(rep(1,cases), rep(0,controls))

 # genotype matrix with 10 variants (random data) set.seed(1234)

 genotype = matrix(rbinom(total*10, 2, 0.051), nrow=total, ncol=10)

 # apply CMC with maf=0.05 and 500 permutations

mycmc = CMC(phenotype, genotype, maf=0.05, perm=500) mycmc

 CMC: Combined Multivariate and Collapsing Method 

Info:

   cases   controls   variants  
     500       500        10  
 rarevar       maf      perm  
       5      0.05       500  

 cmc.stat   asym.pval  perm.pval   
 1.767877   0.940315    0.940000

##### **Adaptive Burden tests:** 

Strengths – are stronger than burden tests using fixed weights or thresholds. Some tests can improve result interpretation

Limitations – are usually computationally intensive. VT requires the same assumptions as burden tests

* **aSum test** 

(a) Idea: The aSum test is also called an Adaptive sum test and it is the approach behind the adaptive test is to use the U-statistics of the score test in order to construct a statistic with the first components of the score vector U. 

(b) Model: The Sum test can be used to summarize information on multiple Nucleotide Polymorphisms (SNP’s) with a DF=1. Adaptive Sum tests are low power and have limited control over error type I.

(c) R function

$$aSum=(U-U_0)'V_0^{-1}(U-U_0)$$

ASUM(y, X, perm = 100)

Y: numeric vector with phenotype status: 0=controls, 1=cases. No missing data allowed

X: numeric matrix or data frame with genotype data coded as 0, 1, 2. Missing data is allowed

(d) Example

## Not run: 

# # number of cases

# cases = 500

# # number of controls

# controls = 500

# # total (cases + controls)

# total = cases + controls

# # phenotype vector

# phenotype = c(rep(1, cases), rep(0, controls))

# # genotype matrix with 10 variants (random data) set.seed(123)

# genotype = matrix(rbinom(total*10, 2, 0.05), nrow=total, ncol=10)

# # apply ASUM with 500 permutations

# myasum = ASUM(phenotype, genotype, perm=500) myasum

# # apply ASUM.Ord with 500 permutations

# myasumord = ASSU.Ord(phenotype, genotype, perm=500) myasumord
  
## End(Not run)

ASUM: Adaptive Sum Test 

Info: 
   cases  controls  variants  
     500       500        10  
 n.perms  
     500  

asum.stat   perm.pval   
-1.167671    0.650000 

##### **Variance-Component tests:** 

Strengths – are powerful in the presence of both trait increasing and decreasing variants or a small amount of causal variants

Limitations – are less powerful than burden tests when most variants are causal and effects are in the same direction

* **SKAT** 

(a) Idea: SKAT is a Sequencing Kernal Association Test which can be used to calculate p-values by fitting the null mode containing only the covariates and so it can easily be applied to genome-wide data.

(b) Model: SKAT tests have a relationship between variants and the phenotype by using classical multiple linear and logistic regression.

(c) R function
 
$$Q_{SKAT}=\sum_{j=1}^{m}w_j^2S_j^2$$

SKAT(y, X, kernel = "linear", weights = NULL, a = 1, b = 25, perm = NULL) 

Y: numeric vector with phenotype status: 0=controls, 1=cases. No missing data allowed

X: numeric matrix or data frame with genotype data coded as 0, 1, 2.

(d) Examples

## Not run: 
   
# # load data genodata

# data(genodata)

# # phenotype (first column of genodata)

# pheno = genodata[,1]

# # genotype (rest of columns of genodata)

# geno = genodata[,-1]

# # apply SKAT with linear kernel 

# myskat.linear = SKAT(pheno, geno, kernel="linear") myskat.linear

# # apply SKAT with weighted linear kernel

# # weights estimated from distribution beta(MAF, a=1, b=25)

# myskat.wlinear = SKAT(pheno, geno, kernel="wlinear", a=1, b=25) myskat.wlinear

# # apply SKAT with quadratic kernel

# myskat.quad = SKAT(pheno, geno, kernel="quadratic") myskat.quad

# # apply SKAT with IBS kernel

# myskat.ibs = SKAT(pheno, geno, kernel="IBS") myskat.ibs
  
## End(Not run)

SKAT: Sequence Kernel Association Test 

Info: 
   cases  controls  variants  
    1000      1000        20  
 n.perms    kernel  
    NULL    linear  

skat.stat   asym.pval   perm.pval   
571.75000     0.26249          NA  

SKAT: Sequence Kernel Association Test 

Info: 
   cases  controls  variants  
    1000      1000        20  
 n.perms    kernel  
    NULL   wlinear  

   skat.stat      asym.pval   
27834.518115       0.345633   
   perm.pval   
          NA 
          
 SKAT: Sequence Kernel Association Test 

Info: 
    cases   controls   variants  
     1000       1000         20  
  n.perms     kernel  
     NULL  quadratic  

  skat.stat     asym.pval   
2894.750000      0.351836   
  perm.pval   
         NA          

SKAT: Sequence Kernel Association Test 

Info: 
   cases  controls  variants  
    1000      1000        20  
 n.perms    kernel  
    NULL       IBS  

skat.stat   asym.pval   perm.pval   
 25.65000     0.31155          NA 

* **C-alpha test** 

(a) Idea: C-alpha tests allow for consistent and/or increased power and straightforward calculations when testing the variance of a set containing both protective and risk variants.

(b) Model: C-alpha is used to find patterns across a whole set of rare variants in a certain region. The distribution of counts follows a binomial distribution if alleles do not associate with any phenotype in the targeted region.

(c) R function

$$T=\sum_{i=1}^{m}[(y_i - n_ip_0)^2 - n_ip_0(1-p_0)]$$

$$c=\sum_{n=2}^{max   n}m(n)\sum_{u=0}^{n}[(u-np_0)^2- np_0(1-p_0)]^2f(u|n,p_0)$$

CALPHA(y, X, perm = NULL)

Y: numeric vector with phenotype status: 0=controls, 1=cases. No missing data allowed

X: numeric matrix or data frame with genotype data coded as 0, 1, 2.

(d) Example

## Not run: 
    
# # number of cases

# cases = 500

# # number of controls

# controls = 500
 
# # total (cases + controls)

# total = cases + controls
 
# # phenotype vector

# phenotype = c(rep(1,cases), rep(0,controls))

# # genotype matrix with 10 variants (random data) set.seed(123)

# genotype = matrix(rbinom(total*10, 2, 0.10), nrow=total, ncol=10)
 
# # apply CALPHA with 500 permutations

# mycalpha = CALPHA(phenotype, genotype, perm=500)

# # this is what we get

# mycalpha

## End(Not run)

CALPHA: c-alpha Test 

Info: 
   cases  controls  variants  
     500       500        10
  n.perms  
     500  

calpha.stat     asym.pval   
-348.500000      0.093765   
  perm.pval   
   0.036000 

##### **Combined tests:** 

Strengths – are stronger with respect to the percentage of causal variants and the presence of both trait increasing and decreasing variants.

Limitations – can be slightly less powerful than burden or variance-component tests if their assumptions are largely held; some methods are computationally intensive like the Fisher method

* **SKAT-O** 

(a) Idea: SKAT-O is part of the extended family of SKAT and it is optimal unified test. This test doesn’t need to be resampled when computing asymptotic p values and it is easy to adjust to covariates.

(b) Model: The Qoptimal test, also know as the optimal unified test can be calculated to find the p-value by simply using a one-dimensional numerical integration. When the trait is binary, the sample sizes are small and the p-values have produced more conservative results, then this can lead to incorrect type I error and a loss of power.

(c) R function

$$Q_{optimal}=min_{0<p<1}p_p$$

$$Q_{optimal}=min[p_{p_1},...,p_{p_b}]$$

SKAT(Z, obj, kernel = "linear.weighted", 
  method="davies", weights.beta=c(1,25), weights=NULL, 
  impute.method="fixed", r.corr=0, is_check_genotype=TRUE,
  is_dosage = FALSE, missing_cutoff=0.15 , max_maf=1, estimate_MAF=1)

SKAT.SSD.OneSet(SSD.INFO, SetID, obj, ... ,obj.SNPWeight=NULL)

SKAT.SSD.OneSet_SetIndex(SSD.INFO, SetIndex, obj, ... ,obj.SNPWeight=NULL)

Method="SKATO"

Method="optimal.adj"

(d) Example

data(SKAT.example)

attach(SKAT.example)

## Not run:

# # SKAT with default Beta(1,25) Weights

# # - without covariates

# # continuous trait

# obj<-SKAT_Null_Model(y.c ~ 1, out_type="C")

# SKAT(Z, obj)$p.value

# # dichotomous trait

# obj<-SKAT_Null_Model(y.b ~ 1, out_type="D")

# SKAT(Z, obj)$p.value

# # SKAT with default Beta(1,25) Weights

# # - with covariates

# # continuous trait

# obj<-SKAT_Null_Model(y.c ~ X, out_type="C")

# SKAT(Z, obj)$p.value

# obj.b<-SKAT_Null_Model(y.b ~ X, out_type="D")

# SKAT(Z, obj.b)$p.value

# # SKAT with default Beta(1,25) Weights

# # - Optimal Test

# SKAT(Z, obj, method="optimal.adj")$p.value

# # you can get the same p-value by using method="SKATO"

# SKAT(Z, obj, method="SKATO")$p.value

# # SKAT with Beta(1,30) Weights

# SKAT(Z, obj, weights.beta=c(1,30))$p.value

# dexp(1) - exp(-1) #-> 0

## End(Not run)

Optimal unified test (SKAT-O)

p-value 0.003849451

##### **EC test:** 

Strengths – is powerful when a very small proportion of variants are causal

Limitations – is computationally intensive; is less powerful when a moderate or large proportion of variants are causal

* **EC test** 

(a) Idea: The EC test, standing for exponential combination test, is general rather than a one test statistic and it combines individual variant statistics leading to the making of a set-based analysis. This test leads to powerful combination procedures.

(b) Model: Large exponential weights may lead to low power when distinguishing the alternative from the null. When the alternative is less sparse, then it becomes a little more powerful.

(c) R function

$$Q_{EC}=\sum_{j=1}^{m}exp(\frac{S_j^2}{2var(Sj)})$$

dexp(x, rate = 1, log = FALSE)
pexp(q, rate = 1, lower.tail = TRUE, log.p = FALSE)
qexp(p, rate = 1, lower.tail = TRUE, log.p = FALSE)
rexp(n, rate = 1)

Y:numeric vector with phenotype status: 0=controls, 1=cases. No missing data allowed

X: numeric matrix or data frame with genotype data coded as 0, 1, 2.

(d) Example



Sources:

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4085641/ 

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2842185/

https://www.rdocumentation.org/packages/AssotesteR/versions/0.1-10/topics/CMC 

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2912645/

https://rdrr.io/cran/AssotesteR/man/ASUM.html

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3135811/

https://rdrr.io/cran/AssotesteR/man/SKAT.html

https://rdrr.io/cran/AssotesteR/man/CALPHA.html

https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1001322

https://rdrr.io/cran/SKAT/man/SKAT.html

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3415556/

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3516612/ 
